src/node/secretshare.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "tls/entropy.h"
+
+#include <array>
+#include <fmt/format_header_only.h>
+#include <iostream>
+#include <optional>
+#include <vector>
+
+extern "C"
+{
+#include "tls/randombytes.h"
+
+#include <sss/sss.h>
+}
+
+namespace ccf
+{
+  // The SecretSharing class provides static functions to split a secret into
+  // shares and (re-)combine those shares into the original secret.
+  // The size of the secret to share is fixed (SECRET_TO_SHARE_LENGTH, 64
+  // bytes). It is up to the caller to either shrink the secret if it is too
+  // long. If the secret to split is shorter than SECRET_TO_SHARE_LENGTH bytes,
+  // the caller should ignore the extra bytes.
+  class SecretSharing
+  {
+  public:
+    static constexpr size_t SECRET_TO_SHARE_LENGTH = sss_MLEN;
+    static constexpr size_t SHARE_LENGTH = sss_SHARE_LEN;
+    static constexpr size_t MAX_NUMBER_SHARES = 255; // As per sss documentation
```

Comment:
This is now effectively an upper bound on the number of CCF members, we ought to document that, and probably think about solving it.

Reply:
#851